CommunityShedMVC/src/CommunityShedMVC/Controllers/CommunityController.cs

```diff
@@ -0,0 +1,125 @@
+﻿using CommunityShedMVC.Data;
+using CommunityShedMVC.Models;
+using CommunityShedMVC.ViewModels;
+using System.Collections.Generic;
+using System.Web.Mvc;
+
+namespace CommunityShedMVC.Controllers
+{
+    public class CommunityController : BaseController
+    {
+        public ActionResult Index()
+        {
+            return View();
+        }
+
+        public ActionResult Add()
+        {
+            Community community = new Community();
+            return View(community);
+        }
+
+        [HttpPost]
+        public ActionResult Add(Community community)
+        {
+            if (ModelState.IsValid)
+            {
+                // TODO redirect to details page
+                CommunityShedData.AddCommunity(community, CustomUser.Person.Id);
+                return RedirectToAction("Index", "Home");
+            }
+            return View(community);
+        }
+
+        public ActionResult List()
+        {
+            CommunityListViewModel viewModel = new CommunityListViewModel
+            {
+                Communities = CommunityShedData.GetCommunityListItems(CustomUser.Person.Id)
+            };
+            return View(viewModel);
+        }
+
+        public ActionResult Details(int communityId)
+        {
+            // Querying the database for each property can get expensive.
+            CommunityDetailsViewModel viewModel = new CommunityDetailsViewModel
+            {
+                Community = CommunityShedData.GetCommunity(communityId),
+                PersonRoles = CommunityShedData.GetCommunityPersonRoles(communityId),
+                Members = CommunityShedData.GetCommunityMembers(communityId),
+                CanEdit = CustomUser.CanEditCommunity(communityId),
+                CanEditMembers = CustomUser.IsInRole("Enforcer", communityId)
```

Comment:
I really like the use of boolean properties on the view models to represent the user's permissions and/or roles.